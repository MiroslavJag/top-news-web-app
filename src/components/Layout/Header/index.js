import React from 'react'
import styled from 'styled-components'
import NavigationItems from 'components/Navigation/NavigationItems'
import {useTranslation} from 'react-i18next'

const HeaderWrapper = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  border-bottom: 1px solid lightgrey;
  position: relative;
  width: 100%;
  height: 60px;
  background-color: white;
`
const LanguageSection = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  cursor: pointer;

  p:nth-child(2) {
    cursor: auto;
    &:hover {
      color: darkslategrey;
    }
  }

  p:nth-child(3) {
    margin-right: 20px;
  }
`

const LangItem = styled.p`
  font-size: 14px;
  margin: 0 5px;
  color: darkslategrey;

  &:hover {
    color: grey;
  }
`

const Header = () => {
  const {t, i18n} = useTranslation()

  const changeLanguage = (code) => {
    i18n.changeLanguage(code)
  }

  return (
    <HeaderWrapper>
      <NavigationItems />
      <LanguageSection>
        <LangItem onClick={() => changeLanguage('gb')}>GB</LangItem>
        <LangItem>|</LangItem>
        <LangItem onClick={() => changeLanguage('us')}>US</LangItem>
      </LanguageSection>
    </HeaderWrapper>
  )
}

export default Header
