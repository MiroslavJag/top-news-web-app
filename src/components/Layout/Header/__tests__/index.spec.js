import React from 'react'
import {mount} from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Header from 'components/Layout/Header'
import {BrowserRouter} from 'react-router-dom'

describe('<Header />', () => {
  it('match snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Header t={(k) => k} />
        </BrowserRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should render correctly with all needed components', () => {
    const wrapper = mount(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    )
    expect(wrapper.find('NavigationItems__NavigationList')).toHaveLength(1)
    expect(wrapper.find('Header__LanguageSection')).toHaveLength(1)
  })
})
