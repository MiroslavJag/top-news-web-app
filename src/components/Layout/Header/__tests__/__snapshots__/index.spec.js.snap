// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> match snapshot 1`] = `
.c2 {
  padding: 0 20px;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: large;
  color: darkslategrey;
}

.c3:hover {
  color: grey;
}

.c3.active {
  color: darkcyan;
}

.c1 {
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0;
  margin: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid lightgrey;
  position: relative;
  width: 100%;
  height: 60px;
  background-color: white;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  cursor: pointer;
}

.c4 p:nth-child(2) {
  cursor: auto;
}

.c4 p:nth-child(2):hover {
  color: darkslategrey;
}

.c4 p:nth-child(3) {
  margin-right: 20px;
}

.c5 {
  font-size: 14px;
  margin: 0 5px;
  color: darkslategrey;
}

.c5:hover {
  color: grey;
}

<div
  className="c0"
>
  <ul
    className="c1"
  >
    <li
      className="c2"
    >
      <a
        aria-current="page"
        className="c3 active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Top News
      </a>
    </li>
    <li
      className="c2"
    >
      <a
        aria-current={null}
        className="c3"
        href="/categories"
        onClick={[Function]}
      >
        Categories
      </a>
    </li>
    <li
      className="c2"
    >
      <a
        aria-current={null}
        className="c3"
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </li>
  </ul>
  <div
    className="c4"
  >
    <p
      className="c5"
      onClick={[Function]}
    >
      GB
    </p>
    <p
      className="c5"
    >
      |
    </p>
    <p
      className="c5"
      onClick={[Function]}
    >
      US
    </p>
  </div>
</div>
`;
